File: fit/src/main/java/com/dreampany/demo/ui/fragment/SettingsFragment.java
package com.dreampany.demo.ui.fragment;

import android.os.Bundle;

import com.dreampany.demo.R;
import com.dreampany.demo.data.source.room.Pref;
import com.dreampany.demo.service.NotifyService;
import com.dreampany.frame.api.service.ServiceManager;
import com.dreampany.frame.misc.ActivityScope;
import com.dreampany.frame.misc.AppExecutors;
import com.dreampany.frame.misc.RxMapper;
import com.dreampany.frame.ui.fragment.BaseMenuFragment;

import org.jetbrains.annotations.Nullable;

import javax.inject.Inject;

import io.reactivex.Flowable;
import io.reactivex.disposables.Disposable;
import timber.log.Timber;

/**
 * Created by Hawladar Roman on 5/28/2018.
 * BJIT Group
 * hawladar.roman@bjitgroup.com
 */
@ActivityScope
public class SettingsFragment extends BaseMenuFragment {

    @Inject
    Pref pref;
    @Inject
    RxMapper rx;
    @Inject
    AppExecutors ex;
    @Inject
    ServiceManager service;
    Disposable disposable;

    @Inject
    public SettingsFragment() {

    }

    @Override
    public int getPrefLayoutId() {
        return R.xml.settings;
    }

    @Override
    protected void onStartUi(@Nullable Bundle state) {
        ex.postToUi(this::initView);
    }

    @Override
    protected void onStopUi() {
        if (disposable != null && !disposable.isDisposed()) {
            disposable.isDisposed();
        }
    }

    private void initView() {
        setTitle(R.string.settings);
        String notifyKey = getString(R.string.key_notification);
        Flowable<Boolean> flowable = pref.observePublic(notifyKey, Boolean.class, true);
        Disposable disposable = rx
                .backToMain(flowable)
                .subscribe(enabled -> {
                    Timber.v("Notification %s" , enabled);
                    if (enabled) {
                        service.schedulePowerService(NotifyService.class, 10);
                    } else {
                        service.cancel(NotifyService.class);
                    }
                });
        this.disposable = disposable;
    }
}

File: word/src/main/kotlin/com/dreampany/word/ui/fragment/SettingsFragment.kt
package com.dreampany.word.ui.fragment

import android.os.Bundle
import com.dreampany.frame.api.service.ServiceManager
import com.dreampany.frame.misc.ActivityScope
import com.dreampany.frame.misc.RxMapper
import com.dreampany.frame.ui.fragment.BaseMenuFragment
import com.dreampany.word.R
import com.dreampany.word.data.source.pref.Pref
import com.dreampany.word.misc.Constants
import com.dreampany.word.service.NotifyService
import io.reactivex.disposables.CompositeDisposable
import javax.inject.Inject

/**
 * Created by Roman-372 on 7/29/2019
 * Copyright (c) 2019 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */
@ActivityScope
class SettingsFragment @Inject constructor() : BaseMenuFragment() {

    @Inject
    internal lateinit var rx: RxMapper
    @Inject
    internal lateinit var service: ServiceManager
    @Inject
    internal lateinit var pref: Pref
    private val disposables: CompositeDisposable

    init {
        disposables = CompositeDisposable()
    }

    override fun getPrefLayoutId(): Int {
        return R.xml.settings
    }

    override fun getScreen(): String {
        return Constants.settings(context!!)
    }

    override fun onStartUi(state: Bundle?) {
        ex.postToUi { this.initView() }
    }

    override fun onStopUi() {
        disposables.clear()
    }

    private fun initView() {
        setTitle(R.string.settings)
        val notifyWordSync = getString(R.string.key_word_sync)
        val eventFlowable = pref.observePublicly(notifyWordSync, Boolean::class.java, true)

        disposables.add(rx
            .backToMain(eventFlowable)
            .subscribe { enabled -> adjustNotify() })
    }

    private val runner = Runnable { this.configJob() }

    private fun configJob() {
        if (pref.hasNotification()) {
            service.schedulePowerService(NotifyService::class.java, Constants.Time.NotifyPeriod.toInt())
        } else {
            service.cancel(NotifyService::class.java)
        }
    }

    private fun adjustNotify() {
        ex.postToUi(runner, 2000)
    }
}